core-container/entry.sh

```diff
@@ -43,56 +33,78 @@ function set_provider_adapters() {
     rm kcpsecret
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl delete spaceproviderdesc $PROVIDER_NAME ; then
         echo "Nothing to delete."
     fi
     kubectl create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+}
  
-        echo "Delete the kubeflex provider object if it already exists, and then create it."
-        if ! kubectl delete spaceproviderdesc pkflex ; then
-            echo "Nothing to delete."
-        fi
-        kubectl create -f - <<EOF
+function create_kubeflex_provider() {
+    echo "Waiting for the kubeflex provider to be ready... this may take a while."
+    (
+        until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
+            sleep 10
+        done
+    )
+
+    echo "Delete the kubeflex provider object if it already exists, and then create it."
+    if ! kubectl delete spaceproviderdesc $PROVIDER_NAME ; then
+        echo "Nothing to delete."
+    fi
+    kubectl create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkflex
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kubeflex"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: corecluster
 EOF
-    fi
+    echo "Waiting for default spaceprovider to reach the Ready phase."
+    until [ "$(kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get spaceproviderdesc $PROVIDER_NAME -o yaml | grep Ready )" != "" ]; do
+        sleep 1
+    done 
+}
+
 
+function set_provider_adapters() {
```

Comment:
I do not understand why this function is named "set_provider_adapters"; I see no adapters here.